packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -480,10 +561,40 @@ export class CdkToolkit {
     return assembly.stackById(stacks.firstStack.id);
   }
 
-  private assembly(): Promise<CloudAssembly> {
-    return this.props.cloudExecutable.synthesize();
+  private assembly(cacheCloudAssembly?: boolean): Promise<CloudAssembly> {
+    return this.props.cloudExecutable.synthesize(cacheCloudAssembly);
+  }
+
+  private patternsArrayForWatch(patterns: string | string[] | undefined, rootDir: string, returnRootDirIfEmpty: boolean): string[] {
```

Comment:
use a struct to make call sites more readable:

```ts
createWatchPattern(patterns: string | string[], options: {
  rootDir: string;
  returnRootIfEmpty: boolean;
});

Reply:
Done!